### ISSUE TITLE
owlbot: strange behavior on regenerating a merged PR

### ISSUE THREAD

**Issue by bshaffer** [googleapis/repo-automation-bots#3068]
I merged an OwlBot PR (see https://github.com/googleapis/google-cloud-php/pull/4932), but then I updated the `owlbot.py` for that PR (see https://github.com/googleapis/google-cloud-php/pull/4987) and regenerated it in order to create a new branch and submit a NEW PR with any potential changes which were missed. This resulted in https://github.com/googleapis/google-cloud-php/compare/owl-bot-e51539e8-ddc6-479f-b298-b3ee6bdff434, which is very bizarre as it's _nothing_ like the original PR. It commits the entire `owl-bot-staging` directory.

I'm not sure where the logic is to ignore `owl-bot-staging`, or why this behavior is happening _only_ when a merged PR is regenerated. Any help would be appreciated!

(On a side note, if a PR is merged and regenerated, it would be nice if OwlBot created a _new_ PR as well)

**Comment by bshaffer** [comment]
UPDATE: This happened just on an unmerged PR (which is strange because I've regenerated many PRs without this issue)

https://github.com/googleapis/google-cloud-php/pull/4903

**Comment by SurferJeffAtGoogle** [comment]
> Okay, so I think maybe this is just how owlbot works.... That the staging files are committed intentionally, and the post-processing happens in another step. 

Indeed.

> But if that's the case, it seems that if the PR is closed, the post-processing never happens.

If the PR is closed, I assume the post processing doesn't need to happen.  Are you concerned about what happens if the PR is merged?

**Comment by bshaffer** [comment]
@SurferJeffAtGoogle Yes, exactly. What happened in this case is I merged a PR and **then** realized there may have been an error in the generation and wanted to regenerate it. Is there any possible way to trigger a new build in that case? I had been hoping I could manually resubmit the PR branch, but since the post-processing doesn't take place, that isn't possible.

Another concern is, with the current architecture, we trigger the test-suite twice, which clogs up the kokoro pipeline.

**Comment by SurferJeffAtGoogle** [comment]
> @SurferJeffAtGoogle Yes, exactly. What happened in this case is I merged a PR and **then** realized there may have been an error in the generation and wanted to regenerate it. Is there any possible way to trigger a new build in that case? I had been hoping I could manually resubmit the PR branch, but since the post-processing doesn't take place, that isn't possible.

It can be done manually.  Follow the instructions here:
[owlbot-howto#how-will-i-test-my-githubowlbotyaml-file](https://g3doc.corp.google.com/company/teams/dpe-github-automation/owlbot.md?cl=head#how-will-i-test-my-githubowlbotyaml-file)
and pass an extra `--source-repo-commit-hash` flag to specify the commit hash from googleapis-gen that you want to copy code from.

Then create a PR from the changes and the post processor will run.

> Another concern is, with the current architecture, we trigger the test-suite twice, which clogs up the kokoro pipeline.

That concern was first aired in Owl Bot's design review.  It's not free, but it's not very costly either.
